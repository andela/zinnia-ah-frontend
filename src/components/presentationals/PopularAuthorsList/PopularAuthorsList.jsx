import React from 'react';
import PropTypes from 'prop-types';

import PopularAuthorItem from '../PopularAuthorItem/PopularAuthorItem';
import Title from '../Title/Title';

const PopularAuthorsList = ({ authors }) => {
  return (
    <div className="popular-authors">
      <Title content="Popular Authors" className="title-md index-title" />
      <div className="author-slot-container">
        {authors.map(author => (
          <PopularAuthorItem
            key={author.id}
            name={author.name}
            url={author.url}
            image={author.image}
            username={author.username}
          />
        ))}
      </div>
    </div>
  );
};

PopularAuthorsList.propTypes = {
  authors: PropTypes.array,
};

export default PopularAuthorsList;
