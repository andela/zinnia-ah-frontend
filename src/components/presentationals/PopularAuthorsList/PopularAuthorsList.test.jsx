import React from 'react';
import { mount } from 'enzyme';

import PopularAuthorsList from './PopularAuthorsList';

describe('Popular Authors List', () => {
  it('should render correctly', () => {
    const component = mount(<PopularAuthorsList authors={[]} className="" />);
    expect(component).toMatchSnapshot();
  });
});
