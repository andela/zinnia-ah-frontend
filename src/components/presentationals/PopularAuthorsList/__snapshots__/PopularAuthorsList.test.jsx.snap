// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular Authors List should render correctly 1`] = `
<PopularAuthorsList
  authors={Array []}
  className=""
>
  <div
    className="popular-authors"
  >
    <Title
      className="title-md index-title"
      content="Popular Authors"
    >
      <h1
        className="title title-md index-title"
      >
        Popular Authors
      </h1>
    </Title>
    <div
      className="author-slot-container"
    />
  </div>
</PopularAuthorsList>
`;
